src/main/java/com/northconcepts/datapipeline/foundations/examples/datamapping/DeclarativeDataMapping.java

```diff
@@ -0,0 +1,36 @@
+package com.northconcepts.datapipeline.foundations.examples.datamapping;
+
+import com.northconcepts.datapipeline.core.DataReader;
+import com.northconcepts.datapipeline.core.StreamWriter;
+import com.northconcepts.datapipeline.csv.CSVReader;
+import com.northconcepts.datapipeline.foundations.datamapping.DataMapping;
+import com.northconcepts.datapipeline.foundations.datamapping.DataMappingReader;
+import com.northconcepts.datapipeline.foundations.schema.EntityDef;
+import com.northconcepts.datapipeline.foundations.schema.SchemaDef;
+import com.northconcepts.datapipeline.job.Job;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.InputStream;
+
+public class DeclarativeDataMapping {
+
+    public static void main(String... args) throws Throwable {
+        DataReader reader = new CSVReader(new File("example/data/input/credit-balance-02.csv"))
+            .setFieldNamesInFirstRow(true);
+
+        File schemaXmlFile = new File("example/data/input/account-schema-definition.xml");
+        SchemaDef schema = new SchemaDef().fromXml(new FileInputStream(schemaXmlFile));
+        EntityDef entityDef = schema.getEntity("Account");
+
+        InputStream stream = new FileInputStream("example/data/input/credit-balance-mapping.xml");
+
+        DataMapping mapping = new DataMapping();
+        mapping.setTargetValidationEntity(entityDef);
```

Comment:
The `entityDef` is for the source/incoming data, let's drop it from here and create a different/separate entity for the target (in same schema file).  Let's also give it a descriptive name that includes its intent (like `sourceAccountEntity`).

Reply:
done